go-controller/pkg/ovn/pods.go

```diff
@@ -285,6 +285,13 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 		opts = make(map[string]string)
 	}
 	opts["requested-chassis"] = pod.Spec.NodeName
+	// Unique identifier to distinguish interfaces for recreated pods, also set by ovnkube-node
+	// ovn-controller will claim the OVS interface only if external_ids:iface-id
+	// matches with the Port_Binding.logical_port and external_ids:iface-id-ver matches
+	// with the Port_Binding.options:iface-id-ver. This is not mandatory.
+	// If Port_binding.options:iface-id-ver is not set, then OVS
+	// Interface.external_ids:iface-id-ver if set is ignored.
+	opts["iface-id-ver"] = string(pod.UID)
```

Comment:
per our slack discussion, just move this up to the if lsp==nil block on 266. Then only new pods will have this set. We dont need to worry about upgrade because ovn-controller is backwards compatible if it doesnt see the new option on the LSP it will bind without checking for it in ovsdb column.